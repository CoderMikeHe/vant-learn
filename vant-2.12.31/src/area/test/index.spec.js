import Area from '..';
import areaList from '../demo/area-simple';
import { mount, later, triggerDrag } from '../../../test';

const firstOption = [
  { code: '110000', name: '北京市' },
  { code: '110100', name: '北京市' },
  { code: '110101', name: '东城区' },
];

const secondOption = [
  { code: '120000', name: '天津市' },
  { code: '120100', name: '天津市' },
  { code: '120101', name: '和平区' },
];

test('confirm & cancel event', async () => {
  const onConfirm = jest.fn();
  const onCancel = jest.fn();
  const wrapper = mount(Area, {
    propsData: {
      areaList,
    },
    listeners: {
      confirm: onConfirm,
      cancel: onCancel,
    },
  });

  await later();

  wrapper.find('.van-picker__confirm').trigger('click');
  wrapper.find('.van-picker__cancel').trigger('click');

  expect(onConfirm).toHaveBeenCalledWith(firstOption, [0, 0, 0]);
  expect(onCancel).toHaveBeenCalledWith(firstOption, [0, 0, 0]);
});

test('watch areaList & code', async () => {
  const wrapper = mount(Area, {
    propsData: {
      areaList,
    },
  });

  expect(wrapper).toMatchSnapshot();
  wrapper.setProps({ value: '110117' });

  await later();
  expect(wrapper).toMatchSnapshot();

  wrapper.setProps({
    value: '',
  });
  expect(wrapper).toMatchSnapshot();
});

test('change option', () => {
  const onChange = jest.fn();
  const wrapper = mount(Area, {
    propsData: {
      areaList,
    },
    listeners: {
      change: onChange,
    },
  });

  const columns = wrapper.findAll('.van-picker-column');
  expect(wrapper).toMatchSnapshot();

  triggerDrag(columns.at(0), 0, -100);
  columns.at(0).find('ul').trigger('transitionend');
  expect(wrapper).toMatchSnapshot();

  triggerDrag(columns.at(2), 0, -100);
  columns.at(2).find('ul').trigger('transitionend');
  expect(wrapper).toMatchSnapshot();

  expect(onChange.mock.calls[0][1]).toEqual(secondOption);
});

test('getValues method', () => {
  const wrapper = mount(Area, {
    propsData: {
      areaList,
    },
    created() {
      expect(this.getValues()).toEqual([]);
    },
  });

  expect(wrapper.vm.getValues()).toEqual(firstOption);
});

test('reset method', async () => {
  const wrapper = mount(Area, {
    propsData: {
      areaList,
      value: '120225',
    },
  });

  await later();
  expect(wrapper).toMatchSnapshot();
  wrapper.vm.reset();
  expect(wrapper).toMatchSnapshot();
});

test('columns-num prop', async () => {
  const wrapper = mount(Area, {
    propsData: {
      areaList,
      columnsNum: 3,
    },
  });

  wrapper.setProps({
    columnsNum: 2,
  });

  await later();
  expect(wrapper).toMatchSnapshot();
});

test('title slot', async () => {
  const wrapper = mount(Area, {
    scopedSlots: {
      title: () => 'Custom Title',
    },
  });

  await later();
  expect(wrapper).toMatchSnapshot();
});

test('columns-top、columns-bottom slot', async () => {
  const wrapper = mount(Area, {
    slots: {
      'columns-top': 'Top',
      'columns-bottom': 'Bottom',
    },
  });

  await later();
  expect(wrapper).toMatchSnapshot();
});
