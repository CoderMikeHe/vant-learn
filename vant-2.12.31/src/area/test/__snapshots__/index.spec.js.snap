// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`change option 1`] = `
<div class="van-picker van-area">
  <div class="van-picker__toolbar"><button type="button" class="van-picker__cancel">取消</button><button type="button" class="van-picker__confirm">确认</button></div>
  <!---->
  <div class="van-picker__columns" style="height: 264px;">
    <div class="van-picker-column">
      <ul class="van-picker-column__wrapper" style="transform: translate3d(0, 110px, 0); transition-duration: 0ms; transition-property: none;">
        <li role="button" tabindex="0" class="van-picker-column__item van-picker-column__item--selected" style="height: 44px;">
          <div class="van-ellipsis">北京市</div>
        </li>
        <li role="button" tabindex="0" class="van-picker-column__item" style="height: 44px;">
          <div class="van-ellipsis">天津市</div>
        </li>
      </ul>
    </div>
    <div class="van-picker-column">
      <ul class="van-picker-column__wrapper" style="transform: translate3d(0, 110px, 0); transition-duration: 0ms; transition-property: none;">
        <li role="button" tabindex="0" class="van-picker-column__item van-picker-column__item--selected" style="height: 44px;">
          <div class="van-ellipsis">北京市</div>
        </li>
        <li role="button" tabindex="0" class="van-picker-column__item" style="height: 44px;">
          <div class="van-ellipsis">县</div>
        </li>
      </ul>
    </div>
    <div class="van-picker-column">
      <ul class="van-picker-column__wrapper" style="transform: translate3d(0, 110px, 0); transition-duration: 0ms; transition-property: none;">
        <li role="button" tabindex="0" class="van-picker-column__item van-picker-column__item--selected" style="height: 44px;">
          <div class="van-ellipsis">东城区</div>
        </li>
        <li role="button" tabindex="0" class="van-picker-column__item" style="height: 44px;">
          <div class="van-ellipsis">西城区</div>
        </li>
      </ul>
    </div>
    <div class="van-picker__mask" style="background-size: 100% 110px;"></div>
    <div class="van-hairline-unset--top-bottom van-picker__frame" style="height: 44px;"></div>
  </div>
  <!---->
</div>
`;

exports[`change option 2`] = `
<div class="van-picker van-area">
  <div class="van-picker__toolbar"><button type="button" class="van-picker__cancel">取消</button><button type="button" class="van-picker__confirm">确认</button></div>
  <!---->
  <div class="van-picker__columns" style="height: 264px;">
    <div class="van-picker-column">
      <ul class="van-picker-column__wrapper" style="transform: translate3d(0, 66px, 0); transition-duration: 0ms; transition-property: none;">
        <li role="button" tabindex="0" class="van-picker-column__item" style="height: 44px;">
          <div class="van-ellipsis">北京市</div>
        </li>
        <li role="button" tabindex="0" class="van-picker-column__item van-picker-column__item--selected" style="height: 44px;">
          <div class="van-ellipsis">天津市</div>
        </li>
      </ul>
    </div>
    <div class="van-picker-column">
      <ul class="van-picker-column__wrapper" style="transform: translate3d(0, 110px, 0); transition-duration: 0ms; transition-property: none;">
        <li role="button" tabindex="0" class="van-picker-column__item van-picker-column__item--selected" style="height: 44px;">
          <div class="van-ellipsis">天津市</div>
        </li>
        <li role="button" tabindex="0" class="van-picker-column__item" style="height: 44px;">
          <div class="van-ellipsis">县</div>
        </li>
      </ul>
    </div>
    <div class="van-picker-column">
      <ul class="van-picker-column__wrapper" style="transform: translate3d(0, 110px, 0); transition-duration: 0ms; transition-property: none;">
        <li role="button" tabindex="0" class="van-picker-column__item van-picker-column__item--selected" style="height: 44px;">
          <div class="van-ellipsis">和平区</div>
        </li>
        <li role="button" tabindex="0" class="van-picker-column__item" style="height: 44px;">
          <div class="van-ellipsis">河东区</div>
        </li>
      </ul>
    </div>
    <div class="van-picker__mask" style="background-size: 100% 110px;"></div>
    <div class="van-hairline-unset--top-bottom van-picker__frame" style="height: 44px;"></div>
  </div>
  <!---->
</div>
`;

exports[`change option 3`] = `
<div class="van-picker van-area">
  <div class="van-picker__toolbar"><button type="button" class="van-picker__cancel">取消</button><button type="button" class="van-picker__confirm">确认</button></div>
  <!---->
  <div class="van-picker__columns" style="height: 264px;">
    <div class="van-picker-column">
      <ul class="van-picker-column__wrapper" style="transform: translate3d(0, 66px, 0); transition-duration: 0ms; transition-property: none;">
        <li role="button" tabindex="0" class="van-picker-column__item" style="height: 44px;">
          <div class="van-ellipsis">北京市</div>
        </li>
        <li role="button" tabindex="0" class="van-picker-column__item van-picker-column__item--selected" style="height: 44px;">
          <div class="van-ellipsis">天津市</div>
        </li>
      </ul>
    </div>
    <div class="van-picker-column">
      <ul class="van-picker-column__wrapper" style="transform: translate3d(0, 110px, 0); transition-duration: 0ms; transition-property: none;">
        <li role="button" tabindex="0" class="van-picker-column__item van-picker-column__item--selected" style="height: 44px;">
          <div class="van-ellipsis">天津市</div>
        </li>
        <li role="button" tabindex="0" class="van-picker-column__item" style="height: 44px;">
          <div class="van-ellipsis">县</div>
        </li>
      </ul>
    </div>
    <div class="van-picker-column">
      <ul class="van-picker-column__wrapper" style="transform: translate3d(0, 66px, 0); transition-duration: 0ms; transition-property: none;">
        <li role="button" tabindex="0" class="van-picker-column__item" style="height: 44px;">
          <div class="van-ellipsis">和平区</div>
        </li>
        <li role="button" tabindex="0" class="van-picker-column__item van-picker-column__item--selected" style="height: 44px;">
          <div class="van-ellipsis">河东区</div>
        </li>
      </ul>
    </div>
    <div class="van-picker__mask" style="background-size: 100% 110px;"></div>
    <div class="van-hairline-unset--top-bottom van-picker__frame" style="height: 44px;"></div>
  </div>
  <!---->
</div>
`;

exports[`columns-num prop 1`] = `
<div class="van-picker van-area">
  <div class="van-picker__toolbar"><button type="button" class="van-picker__cancel">取消</button><button type="button" class="van-picker__confirm">确认</button></div>
  <!---->
  <div class="van-picker__columns" style="height: 264px;">
    <div class="van-picker-column">
      <ul class="van-picker-column__wrapper" style="transform: translate3d(0, 110px, 0); transition-duration: 0ms; transition-property: none;">
        <li role="button" tabindex="0" class="van-picker-column__item van-picker-column__item--selected" style="height: 44px;">
          <div class="van-ellipsis">北京市</div>
        </li>
        <li role="button" tabindex="0" class="van-picker-column__item" style="height: 44px;">
          <div class="van-ellipsis">天津市</div>
        </li>
      </ul>
    </div>
    <div class="van-picker-column">
      <ul class="van-picker-column__wrapper" style="transform: translate3d(0, 110px, 0); transition-duration: 0ms; transition-property: none;">
        <li role="button" tabindex="0" class="van-picker-column__item van-picker-column__item--selected" style="height: 44px;">
          <div class="van-ellipsis">北京市</div>
        </li>
        <li role="button" tabindex="0" class="van-picker-column__item" style="height: 44px;">
          <div class="van-ellipsis">县</div>
        </li>
      </ul>
    </div>
    <div class="van-picker__mask" style="background-size: 100% 110px;"></div>
    <div class="van-hairline-unset--top-bottom van-picker__frame" style="height: 44px;"></div>
  </div>
  <!---->
</div>
`;

exports[`columns-top、columns-bottom slot 1`] = `
<div class="van-picker van-area">
  <div class="van-picker__toolbar"><button type="button" class="van-picker__cancel">取消</button><button type="button" class="van-picker__confirm">确认</button></div>
  <!---->Top<div class="van-picker__columns" style="height: 264px;">
    <div class="van-picker-column">
      <ul class="van-picker-column__wrapper" style="transform: translate3d(0, 110px, 0); transition-duration: 0ms; transition-property: none;"></ul>
    </div>
    <div class="van-picker-column">
      <ul class="van-picker-column__wrapper" style="transform: translate3d(0, 110px, 0); transition-duration: 0ms; transition-property: none;"></ul>
    </div>
    <div class="van-picker-column">
      <ul class="van-picker-column__wrapper" style="transform: translate3d(0, 110px, 0); transition-duration: 0ms; transition-property: none;"></ul>
    </div>
    <div class="van-picker__mask" style="background-size: 100% 110px;"></div>
    <div class="van-hairline-unset--top-bottom van-picker__frame" style="height: 44px;"></div>
  </div>Bottom
  <!---->
</div>
`;

exports[`reset method 1`] = `
<div class="van-picker van-area">
  <div class="van-picker__toolbar"><button type="button" class="van-picker__cancel">取消</button><button type="button" class="van-picker__confirm">确认</button></div>
  <!---->
  <div class="van-picker__columns" style="height: 264px;">
    <div class="van-picker-column">
      <ul class="van-picker-column__wrapper" style="transform: translate3d(0, 66px, 0); transition-duration: 0ms; transition-property: none;">
        <li role="button" tabindex="0" class="van-picker-column__item" style="height: 44px;">
          <div class="van-ellipsis">北京市</div>
        </li>
        <li role="button" tabindex="0" class="van-picker-column__item van-picker-column__item--selected" style="height: 44px;">
          <div class="van-ellipsis">天津市</div>
        </li>
      </ul>
    </div>
    <div class="van-picker-column">
      <ul class="van-picker-column__wrapper" style="transform: translate3d(0, 66px, 0); transition-duration: 0ms; transition-property: none;">
        <li role="button" tabindex="0" class="van-picker-column__item" style="height: 44px;">
          <div class="van-ellipsis">天津市</div>
        </li>
        <li role="button" tabindex="0" class="van-picker-column__item van-picker-column__item--selected" style="height: 44px;">
          <div class="van-ellipsis">县</div>
        </li>
      </ul>
    </div>
    <div class="van-picker-column">
      <ul class="van-picker-column__wrapper" style="transform: translate3d(0, 110px, 0); transition-duration: 0ms; transition-property: none;">
        <li role="button" tabindex="0" class="van-picker-column__item van-picker-column__item--selected" style="height: 44px;">
          <div class="van-ellipsis">蓟县</div>
        </li>
      </ul>
    </div>
    <div class="van-picker__mask" style="background-size: 100% 110px;"></div>
    <div class="van-hairline-unset--top-bottom van-picker__frame" style="height: 44px;"></div>
  </div>
  <!---->
</div>
`;

exports[`reset method 2`] = `
<div class="van-picker van-area">
  <div class="van-picker__toolbar"><button type="button" class="van-picker__cancel">取消</button><button type="button" class="van-picker__confirm">确认</button></div>
  <!---->
  <div class="van-picker__columns" style="height: 264px;">
    <div class="van-picker-column">
      <ul class="van-picker-column__wrapper" style="transform: translate3d(0, 110px, 0); transition-duration: 0ms; transition-property: none;">
        <li role="button" tabindex="0" class="van-picker-column__item van-picker-column__item--selected" style="height: 44px;">
          <div class="van-ellipsis">北京市</div>
        </li>
        <li role="button" tabindex="0" class="van-picker-column__item" style="height: 44px;">
          <div class="van-ellipsis">天津市</div>
        </li>
      </ul>
    </div>
    <div class="van-picker-column">
      <ul class="van-picker-column__wrapper" style="transform: translate3d(0, 110px, 0); transition-duration: 0ms; transition-property: none;">
        <li role="button" tabindex="0" class="van-picker-column__item van-picker-column__item--selected" style="height: 44px;">
          <div class="van-ellipsis">北京市</div>
        </li>
        <li role="button" tabindex="0" class="van-picker-column__item" style="height: 44px;">
          <div class="van-ellipsis">县</div>
        </li>
      </ul>
    </div>
    <div class="van-picker-column">
      <ul class="van-picker-column__wrapper" style="transform: translate3d(0, 110px, 0); transition-duration: 0ms; transition-property: none;">
        <li role="button" tabindex="0" class="van-picker-column__item van-picker-column__item--selected" style="height: 44px;">
          <div class="van-ellipsis">东城区</div>
        </li>
        <li role="button" tabindex="0" class="van-picker-column__item" style="height: 44px;">
          <div class="van-ellipsis">西城区</div>
        </li>
      </ul>
    </div>
    <div class="van-picker__mask" style="background-size: 100% 110px;"></div>
    <div class="van-hairline-unset--top-bottom van-picker__frame" style="height: 44px;"></div>
  </div>
  <!---->
</div>
`;

exports[`title slot 1`] = `
<div class="van-picker van-area">
  <div class="van-picker__toolbar"><button type="button" class="van-picker__cancel">取消</button>Custom Title<button type="button" class="van-picker__confirm">确认</button></div>
  <!---->
  <div class="van-picker__columns" style="height: 264px;">
    <div class="van-picker-column">
      <ul class="van-picker-column__wrapper" style="transform: translate3d(0, 110px, 0); transition-duration: 0ms; transition-property: none;"></ul>
    </div>
    <div class="van-picker-column">
      <ul class="van-picker-column__wrapper" style="transform: translate3d(0, 110px, 0); transition-duration: 0ms; transition-property: none;"></ul>
    </div>
    <div class="van-picker-column">
      <ul class="van-picker-column__wrapper" style="transform: translate3d(0, 110px, 0); transition-duration: 0ms; transition-property: none;"></ul>
    </div>
    <div class="van-picker__mask" style="background-size: 100% 110px;"></div>
    <div class="van-hairline-unset--top-bottom van-picker__frame" style="height: 44px;"></div>
  </div>
  <!---->
</div>
`;

exports[`watch areaList & code 1`] = `
<div class="van-picker van-area">
  <div class="van-picker__toolbar"><button type="button" class="van-picker__cancel">取消</button><button type="button" class="van-picker__confirm">确认</button></div>
  <!---->
  <div class="van-picker__columns" style="height: 264px;">
    <div class="van-picker-column">
      <ul class="van-picker-column__wrapper" style="transform: translate3d(0, 110px, 0); transition-duration: 0ms; transition-property: none;">
        <li role="button" tabindex="0" class="van-picker-column__item van-picker-column__item--selected" style="height: 44px;">
          <div class="van-ellipsis">北京市</div>
        </li>
        <li role="button" tabindex="0" class="van-picker-column__item" style="height: 44px;">
          <div class="van-ellipsis">天津市</div>
        </li>
      </ul>
    </div>
    <div class="van-picker-column">
      <ul class="van-picker-column__wrapper" style="transform: translate3d(0, 110px, 0); transition-duration: 0ms; transition-property: none;">
        <li role="button" tabindex="0" class="van-picker-column__item van-picker-column__item--selected" style="height: 44px;">
          <div class="van-ellipsis">北京市</div>
        </li>
        <li role="button" tabindex="0" class="van-picker-column__item" style="height: 44px;">
          <div class="van-ellipsis">县</div>
        </li>
      </ul>
    </div>
    <div class="van-picker-column">
      <ul class="van-picker-column__wrapper" style="transform: translate3d(0, 110px, 0); transition-duration: 0ms; transition-property: none;">
        <li role="button" tabindex="0" class="van-picker-column__item van-picker-column__item--selected" style="height: 44px;">
          <div class="van-ellipsis">东城区</div>
        </li>
        <li role="button" tabindex="0" class="van-picker-column__item" style="height: 44px;">
          <div class="van-ellipsis">西城区</div>
        </li>
      </ul>
    </div>
    <div class="van-picker__mask" style="background-size: 100% 110px;"></div>
    <div class="van-hairline-unset--top-bottom van-picker__frame" style="height: 44px;"></div>
  </div>
  <!---->
</div>
`;

exports[`watch areaList & code 2`] = `
<div class="van-picker van-area">
  <div class="van-picker__toolbar"><button type="button" class="van-picker__cancel">取消</button><button type="button" class="van-picker__confirm">确认</button></div>
  <!---->
  <div class="van-picker__columns" style="height: 264px;">
    <div class="van-picker-column">
      <ul class="van-picker-column__wrapper" style="transform: translate3d(0, 110px, 0); transition-duration: 0ms; transition-property: none;">
        <li role="button" tabindex="0" class="van-picker-column__item van-picker-column__item--selected" style="height: 44px;">
          <div class="van-ellipsis">北京市</div>
        </li>
        <li role="button" tabindex="0" class="van-picker-column__item" style="height: 44px;">
          <div class="van-ellipsis">天津市</div>
        </li>
      </ul>
    </div>
    <div class="van-picker-column">
      <ul class="van-picker-column__wrapper" style="transform: translate3d(0, 110px, 0); transition-duration: 0ms; transition-property: none;">
        <li role="button" tabindex="0" class="van-picker-column__item van-picker-column__item--selected" style="height: 44px;">
          <div class="van-ellipsis">北京市</div>
        </li>
        <li role="button" tabindex="0" class="van-picker-column__item" style="height: 44px;">
          <div class="van-ellipsis">县</div>
        </li>
      </ul>
    </div>
    <div class="van-picker-column">
      <ul class="van-picker-column__wrapper" style="transform: translate3d(0, 110px, 0); transition-duration: 0ms; transition-property: none;">
        <li role="button" tabindex="0" class="van-picker-column__item van-picker-column__item--selected" style="height: 44px;">
          <div class="van-ellipsis">东城区</div>
        </li>
        <li role="button" tabindex="0" class="van-picker-column__item" style="height: 44px;">
          <div class="van-ellipsis">西城区</div>
        </li>
      </ul>
    </div>
    <div class="van-picker__mask" style="background-size: 100% 110px;"></div>
    <div class="van-hairline-unset--top-bottom van-picker__frame" style="height: 44px;"></div>
  </div>
  <!---->
</div>
`;

exports[`watch areaList & code 3`] = `
<div class="van-picker van-area">
  <div class="van-picker__toolbar"><button type="button" class="van-picker__cancel">取消</button><button type="button" class="van-picker__confirm">确认</button></div>
  <!---->
  <div class="van-picker__columns" style="height: 264px;">
    <div class="van-picker-column">
      <ul class="van-picker-column__wrapper" style="transform: translate3d(0, 110px, 0); transition-duration: 0ms; transition-property: none;">
        <li role="button" tabindex="0" class="van-picker-column__item van-picker-column__item--selected" style="height: 44px;">
          <div class="van-ellipsis">北京市</div>
        </li>
        <li role="button" tabindex="0" class="van-picker-column__item" style="height: 44px;">
          <div class="van-ellipsis">天津市</div>
        </li>
      </ul>
    </div>
    <div class="van-picker-column">
      <ul class="van-picker-column__wrapper" style="transform: translate3d(0, 110px, 0); transition-duration: 0ms; transition-property: none;">
        <li role="button" tabindex="0" class="van-picker-column__item van-picker-column__item--selected" style="height: 44px;">
          <div class="van-ellipsis">北京市</div>
        </li>
        <li role="button" tabindex="0" class="van-picker-column__item" style="height: 44px;">
          <div class="van-ellipsis">县</div>
        </li>
      </ul>
    </div>
    <div class="van-picker-column">
      <ul class="van-picker-column__wrapper" style="transform: translate3d(0, 110px, 0); transition-duration: 0ms; transition-property: none;">
        <li role="button" tabindex="0" class="van-picker-column__item van-picker-column__item--selected" style="height: 44px;">
          <div class="van-ellipsis">东城区</div>
        </li>
        <li role="button" tabindex="0" class="van-picker-column__item" style="height: 44px;">
          <div class="van-ellipsis">西城区</div>
        </li>
      </ul>
    </div>
    <div class="van-picker__mask" style="background-size: 100% 110px;"></div>
    <div class="van-hairline-unset--top-bottom van-picker__frame" style="height: 44px;"></div>
  </div>
  <!---->
</div>
`;
